src/reportSources/legacyFflogs/eventAdapter/speedStat.ts

```diff
@@ -0,0 +1,252 @@
+import {JobKey} from 'data/JOBS'
+import {getActions, getStatuses} from 'data/layer'
+import {Attribute, Event, Events} from 'event'
+import {BuffEvent, CastEvent, FflogsEvent} from 'fflogs'
+import _ from 'lodash'
+import {Actor, Team} from 'report'
+import {resolveActorId} from 'reportSources/legacyFflogs/base'
+import {GetSpeedStat} from 'utilities/speedStatMapper'
+import {AdapterStep, PREPULL_OFFSETS} from './base'
+
+const BASE_GCD_MS = 2500
+const CASTER_TAX = 0.1
+const JOB_SPEED_MODIFIERS: Partial<Record<JobKey, number>> = {
+	'MONK': 0.8,
+	'NINJA': 0.85,
+}
+interface SpeedmodWindow {
+	start: number,
+	end?: number,
+}
+
+class GCD {
```

Comment:
The current GCD normalizer uses the presence of a cast event without a matching begincast to infer an instant event, rather than taking a dependency on CastTime to tell if the event was instant (due to Swiftcast or what have you), and since this adapter is running where it can't take a dependency on CastTime, we need to maintain that behavior.  As such, I would prefer to let the getters contain the logic of distinguishing which event (prepare or action) was the start of the GCD for the event, rather than having to curate that manually within the event parsing logic.